analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd

```diff
@@ -0,0 +1,641 @@
+
+**Post notes:**   
+It was later determined that the BED files used for these TMB calculations were incorrect See Related Issues:  
+- [568](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/568)  
+- [565](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/565)  
+- [564](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/564)  
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+```{r}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# We will need the calculate_tmb function from this script
+source(file.path("..", "..", "snv-callers", "util", "tmb_functions.R"))
+```
+
+Declare directory paths. 
+
+```{r}
+scratch_dir <- file.path("..", "..", "..", "scratch")
+data_dir <- file.path("..", "..", "..", "data")
+ref_dir <- file.path("..", "ref_files")
+plots_dir <- file.path("plots", "tcga-vs-pbta-plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+### Special functions 
+
+Function for setting up PBTA data from the database. 
+These databases were originally created by the [run consensus bash script for pbta](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/run_caller_consensus_analysis-pbta.sh) and  [run consensus bash script for tcga](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/run_caller_consensus_analysis-tcga.sh)
+
+```{r}
+set_up_data <- function(data_name, 
+                        database_path, 
+                        metadata, 
+                        is_tcga, 
+                        cols_to_keep) {
+  
+  # Given the name of dataset in the database set up the data with the metadata
+  #
+  # Args:
+  #  data_name: A string that is the name of the table in the SQLite file you'd like to pull out.
+  #  database_path: file path to an SQlite file made previously by a run_consensus script
+  #  metadata: associated metadata with this database
+  #  is_tcga: TRUE or FALSE for whether or not this is TCGA data. FALSE = PBTA data
+  #  cols_to_keep: A vector of strings indicating the column names to keep
+  
+  # Start up connection
+  con <- DBI::dbConnect(
+    RSQLite::SQLite(),
+    database_path
+  )
+  # Connect to SQL
+  df <- dplyr::tbl(con, data_name) %>%
+    # Only keep the columns we want
+    dplyr::select(cols_to_keep) %>%
+    # Turn into data.frame
+    as.data.frame() 
+  
+  # This step is only needed for TCGA data
+  if (is_tcga) {
+    df <- df %>% 
+      # Shorten the Tumor_Sample_Barcode so it matches
+      dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12))
+  } 
```

Comment:
I think you can just do this always, and save an argument. Taking the substr of a 12 character barcode will just return the barcode.